contracts/src/contracts/Teleporter/AvalancheICTTRouterEnforcedFees.sol

```diff
@@ -0,0 +1,225 @@
+import {IWarpMessenger} from
+    "@avalabs/subnet-evm-contracts@1.2.0/contracts/interfaces/IWarpMessenger.sol";
+import {Ownable} from "@openzeppelin/contracts@4.8.1/access/Ownable.sol";
+import {ReentrancyGuard} from "@openzeppelin/contracts@4.8.1/security/ReentrancyGuard.sol";
+import {IERC20} from "@openzeppelin/contracts@4.8.1/token/ERC20/IERC20.sol";
+import {SafeERC20} from "@openzeppelin/contracts@4.8.1/token/ERC20/utils/SafeERC20.sol";
+import {Address} from "@openzeppelin/contracts@4.8.1/utils/Address.sol";
+import {SafeERC20TransferFrom} from "@teleporter/SafeERC20TransferFrom.sol";
+
+/// @custom:security-contact security@e36knots.com
+contract AvalancheICTTRouterEnforcedFees is
+    Ownable,
+    ReentrancyGuard,
+    IAvalancheICTTRouterEnforcedFees
+{
+    using Address for address;
+
+    /**
+     * @notice Token address => source bridge address
+     * @notice Address `0x0` is used for the native token
+     */
+    mapping(address token => address sourceBridge) public tokenToSourceBridge;
+
+    /**
+     * @notice Token address => destination chain ID => DestinationBridge
+     * @notice Address `0x0` is used for the native token
+     */
+    mapping(
+        bytes32 destinationChainID => mapping(address token => DestinationBridge destinationBridge)
+    ) public tokenDestinationChainToDestinationBridge;
+
+    /// @notice Relayer fee enforced by the router (in basis points)
+    uint256 public primaryRelayerFeeBips;
+
+    /// @notice Relayer fee enforced by the router (in basis points) in case of multihop bridging during the second bridge
+    uint256 public secondaryRelayerFeeBips;
+
+    /// @notice  Current chain ID
+    bytes32 private immutable routerChainID;
+
+    /**
+     * @notice Set the relayer fee and the ID of the source chain
+     * @param primaryRelayerFeeBips_ Relayer fee in basic points
+     * @param secondaryRelayerFeeBips_ In case of multihop bridge, relayer fee for the second bridge
+     */
+    constructor(uint256 primaryRelayerFeeBips_, uint256 secondaryRelayerFeeBips_) {
+        primaryRelayerFeeBips = primaryRelayerFeeBips_;
+        secondaryRelayerFeeBips = secondaryRelayerFeeBips_;
+        routerChainID = IWarpMessenger(0x0200000000000000000000000000000000000005).getBlockchainID();
+    }
+
+    function setRelayerFeesBips(
+        uint256 primaryRelayerFeeBips_,
+        uint256 secondaryRelayerFeeBips_
+    ) external onlyOwner {
+        primaryRelayerFeeBips = primaryRelayerFeeBips_;
+        secondaryRelayerFeeBips = secondaryRelayerFeeBips_;
+        emit ChangeRelayerFees(primaryRelayerFeeBips_, secondaryRelayerFeeBips_);
+    }
+
+    function registerSourceTokenBridge(
+        address tokenAddress,
+        address bridgeAddress
+    ) external onlyOwner {
+        if (tokenAddress != address(0) && !tokenAddress.isContract()) {
+            revert NotAContract(tokenAddress);
+        }
+        if (!bridgeAddress.isContract()) {
+            revert NotAContract(bridgeAddress);
```

Comment:
I think it might be useful to have 2 errors: `TokenAddrNotAContract` and `BridgeAddrNotAContract`